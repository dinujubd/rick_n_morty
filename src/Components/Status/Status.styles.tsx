import { createStyles, makeStyles, Theme } from "@material-ui/core";
import { PaletteColor } from "@material-ui/core/styles/createPalette";

export const useStyles = (variant: PaletteColor) => makeStyles((theme: Theme) =>
    createStyles({
        root: {
            background: `${variant.dark}`,
            height: '0.7em',
            width: '0.7em',
            borderRadius: '50%',
            display: 'inline-block'
        }
    }),
);