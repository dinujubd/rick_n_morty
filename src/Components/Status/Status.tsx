import React from 'react';
import { Box, Theme, useTheme } from '@material-ui/core';
import { useStyles } from './Status.styles';

interface Props {
    status: string;
}

export const Status: React.FC<Props> = ({ status }) => {

    const getVariant = (status: string, theme: Theme) => {
        switch (status) {
            case 'Alive': return theme.palette.success;
            case 'Dead': return theme.palette.error;
            default: return theme.palette.secondary;
        }
    }
    const theme = useTheme();
    const classes = useStyles(getVariant(status, theme))();

    return (
        <Box component="span">
            <Box component="span" mr={1}>{status}</Box>
            <Box component="span" className={classes.root}></Box>
        </Box>
    );
}

Status.displayName = 'Status'