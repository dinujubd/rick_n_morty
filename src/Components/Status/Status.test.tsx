import renderer from 'react-test-renderer';
import { Status } from './Status';
describe('<Status />', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(<Status status="success"  />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});