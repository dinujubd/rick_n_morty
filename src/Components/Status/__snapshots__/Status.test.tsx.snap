// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> should renders correctly 1`] = `
<span
  className="MuiBox-root MuiBox-root-2"
>
  <span
    className="MuiBox-root MuiBox-root-3"
  >
    success
  </span>
  <span
    className="MuiBox-root MuiBox-root-4 makeStyles-root-1"
  />
</span>
`;
